### PR TITLE
Improve `range_lerp()` and related methods documentation

### PR DESCRIPTION
**Pull request by Calinou**
See https://github.com/godotengine/godot-proposals/issues/5077.

### CODE REVIEW

**Comment by Zireael07** [comment]
Consider adding hints on how to use Gradient for this (I can imagine using a curve, but not gradient)

File: doc/classes/@GlobalScope.xml [hunk 1]
```diff
@@ -380,7 +380,7 @@
...
 			<description>
-				Returns an interpolation or extrapolation factor considering the range specified in [code]from[/code] and [code]to[/code], and the interpolated value specified in [code]weight[/code]. The returned value will be between [code]0.0[/code] and [code]1.0[/code] if [code]weight[/code] is between [code]from[/code] and [code]to[/code] (inclusive). If [code]weight[/code] is located outside this range, then an extrapolation factor will be returned (return value lower than [code]0.0[/code] or greater than [code]1.0[/code]).
+				Returns an interpolation or extrapolation factor considering the range specified in [code]from[/code] and [code]to[/code], and the interpolated value specified in [code]weight[/code]. The returned value will be between [code]0.0[/code] and [code]1.0[/code] if [code]weight[/code] is between [code]from[/code] and [code]to[/code] (inclusive). If [code]weight[/code] is located outside this range, then an extrapolation factor will be returned (return value lower than [code]0.0[/code] or greater than [code]1.0[/code]).  Use [method clamp] on the result of [method inverse_lerp] if this is not desired.
```
Comment (akien-mga) [inline doc/classes/@GlobalScope.xml:383]:
```suggestion
				Returns an interpolation or extrapolation factor considering the range specified in [code]from[/code] and [code]to[/code], and the interpolated value specified in [code]weight[/code]. The returned value will be between [code]0.0[/code] and [code]1.0[/code] if [code]weight[/code] is between [code]from[/code] and [code]to[/code] (inclusive). If [code]weight[/code] is located outside this range, then an extrapolation factor will be returned (return value lower than [code]0.0[/code] or greater than [code]1.0[/code]). Use [method clamp] on the result of [method inverse_lerp] if this is not desired.
```

**Comment by Calinou** [comment]
> Consider adding hints on how to use Gradient for this (I can imagine using a curve, but not gradient)

Gradient is essentially the same use case as Curve, but for colors.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.1.

**Pull request merged**